tests/validators/base.test.ts

```diff
@@ -141,6 +141,29 @@ describe('BaseValidator', () => {
 		});
 	});
 
+	describe('Reshape', () => {
+		const predicate = s.string.reshape((value) => {
+			return value.length > 5
+				? Result.ok(value.length)
+				: // FIXME: if I don't add this number generic, then `reshape` isn't able to infer the type
+				  // Lmk should I keep it like this or r there any fix possible
```

Comment:
@kyranet @vladfrangu

Reply:
shapeshift don't uses `@sapphiredev/result`

Reply:
oh lord. Maybe that's why?

Reply:
Is it worth adding the full package when shapeshift only uses `Ok` and `err`?

Reply:
The idea was for shapeshift to be dependency-less, we've already had to accept some dependencies for the sake of browser compatibility / other validation library parity in `fast-deep-equal` and `lodash.uniqwith` but I would still prefer to keep it at a very low minimum.